src/libs/Localize/index.ts

```diff
@@ -111,33 +134,90 @@ function getTranslatedPhrase<TKey extends TranslationPaths>(
 
     if (translatedPhrase) {
-        if (typeof translatedPhrase === 'function') {
-            return translatedPhrase(...phraseParameters);
-        }
-
-        // We set the translated value in the cache only for the phrases without parameters.
+      // If the translated phrase is a string, cache it and return it
+      if (typeof translatedPhrase === 'string') {
         cacheForLocale?.set(phraseKey, translatedPhrase);
         return translatedPhrase;
+      }
+  
+      let phrase: string | Record<Intl.LDMLPluralRule, string>;
+  
+      // If the translated phrase is a function, evaluate it with the provided parameters
+      if (typeof translatedPhrase === 'function') {
+        phrase = translatedPhrase(...phraseParameters);
+      } else {
+        // If the translated phrase is an object, assign it directly to `phrase`
+        phrase = translatedPhrase as Record<Intl.LDMLPluralRule, string>;
+      }
+  
+      // If `phrase` is a string, return it as the final translated phrase
+      if (typeof phrase === 'string') {
+        return phrase;
+      }
+  
+      // Extract the count from `phraseParameters` (if provided)
+      const count = deserializePhraseParamsForCount(phraseParameters);
+  
+      // Memoize the `Intl.PluralRules` instance for the current language
+      if (!memoizedPluralRules[language]) {
+        memoizedPluralRules[language] = new Intl.PluralRules(language);
+      }
+  
+      // Determine the pluralization category based on the count and language rules
+      const pluralCategory = memoizedPluralRules[language].select(count);
+  
+      // Check if the `phrase` object has a translation for the pluralization category
+    if (Object.prototype.hasOwnProperty.call(phrase, pluralCategory)) {
+        const translation = phrase[pluralCategory];
+        // If the translation is a function, evaluate it with the provided parameters
+        if (typeof translation === 'function') {
+            return translation(...phraseParameters) as string | null;
+        }
+        // Otherwise, return the translation as is
+        return translation;
     }
-
-    if (!fallbackLanguage) {
-        return null;
+  
+      // If no translation is found for the exact category, fallback to the 'other' category
+    if (Object.prototype.hasOwnProperty.call(phrase, 'other')) {
+        const fallbackTranslation = phrase.other;
+        // If the fallback translation is a function, evaluate it with the provided parameters
+        if (typeof fallbackTranslation === 'function') {
+            return fallbackTranslation(...phraseParameters) as string | null;
+        }
+        // Otherwise, return the fallback translation as is
+        const warningMessage = `Plural form '${pluralCategory}' not found for phrase '${phraseKey}'. Using the 'other' category as fallback.`;
+        Log.warn(warningMessage);
+        if (__DEV__) {
+            throw new Error(warningMessage);
+        }
```

Comment:
This should happen in an else of the if starting in line 170. If we don't find the correct one, there's an error, does not matter if we find the `other` one or not.